src/libstd/sys/unix/thread.rs

```diff
@@ -52,6 +52,11 @@ impl Thread {
         assert_eq!(libc::pthread_attr_init(&mut attr), 0);
 
         let stack_size = cmp::max(stack, min_stack_size(&attr));
+
+        // L4Re only supports a maximum of 1Mb per default.
+        #[cfg(target_os = "l4re")]
+        let stack_size = cmp::min(stack_size, 1024 * 1024);
```

Comment:
It will run in an error when pthread_attr_setstacksize is called. (sys/unix/thread.rs line: 55)
Most systems do not have a fixed maximum. We tried to put it to to the point where also the minimum size is checked. Putting it to the pthread_attr_setstacksize check would in my opinion put to much complexity there.